tools/querytee/proxy_endpoint.go

```diff
@@ -218,17 +222,26 @@ func (p *ProxyEndpoint) compareResponses(expectedResponse, actualResponse *backe
 	}
 
 	if expectedResponse.status != actualResponse.status {
-		return fmt.Errorf("expected status code %d but got %d", expectedResponse.status, actualResponse.status)
+		return fmt.Errorf("expected status code %d (returned by preferred backend) but got %d from secondary backend", expectedResponse.status, actualResponse.status)
+	}
+
+	if expectedResponse.contentType != "application/json" {
+		return fmt.Errorf("skipped comparison of response because the response from the preferred backend contained an unexpected content type '%s', expected 'application/json'", expectedResponse.contentType)
```

Comment:
Returning an error here will get logged as "error" level and tracked to `cortex_querytee_responses_compared_total{result="fail"}`. I'm not sure that's correct. We can't compare, so we just don't know if the comparison would have succeeded or failed. I think we shouldn't log it as error and track it in the metric with a new reason, like `skipped`.

Reply:
Makes sense to me, I've changed this in 5d63f01.